sda-commons-server-s3-testing/src/main/java/org/sdase/commons/server/s3/testing/S3ClassExtension.java

```diff
@@ -1,63 +1,125 @@
 package org.sdase.commons.server.s3.testing;
 
+import com.robothy.s3.jupiter.LocalS3;
+import com.robothy.s3.jupiter.LocalS3Endpoint;
+import com.robothy.s3.jupiter.extensions.LocalS3Extension;
 import java.io.File;
 import java.io.InputStream;
+import java.net.URI;
 import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
-import org.junit.jupiter.api.extension.AfterAllCallback;
-import org.junit.jupiter.api.extension.BeforeAllCallback;
+import java.util.Set;
+import java.util.function.Supplier;
 import org.junit.jupiter.api.extension.ExtensionContext;
 import org.sdase.commons.server.s3.testing.builder.ContentObject;
 import org.sdase.commons.server.s3.testing.builder.FileObject;
 import org.sdase.commons.server.s3.testing.builder.MockObject;
 import org.sdase.commons.server.s3.testing.builder.StreamObject;
+import software.amazon.awssdk.auth.credentials.AnonymousCredentialsProvider;
+import software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
 
 /**
- * JUnit 5 extension for running a AWS S3-compatible object storage alongside the (integration)
+ * JUnit 5 extension for running an AWS S3-compatible object storage alongside the (integration)
  * tests. Use {@link #getEndpoint()} to retrieve the endpoint URL to connect to.
  *
  * <p>Example usage:
  *
  * <pre>
+ * &#64;LocalS3
```

Comment:
Question to the reviewers: Do you like this approach?
* good: less code because we extend the Junit 5 extension from local-s3
* bad: `LocalS3` annotation is forced on all our consumers

We could completely re-write the extension by copying the code from local-s3 and adapting it to our needs. Consumers would not need to add the annotation in that case.

Reply:
I would be fine with the annotation.